Sources/ScreenObject/ScreenObject.swift

```diff
@@ -93,15 +93,30 @@ open class ScreenObject {
             activityDescription = "Confirm whole screen \(self) is loaded"
         }
 
-        try XCTContext.runActivity(named: activityDescription) { (activity) in
-            try gettersToTest.forEach { getter in
-                let result = waitFor(
-                    element: getter(app),
-                    predicate: "isHittable == true",
-                    timeout: self.waitTimeout
-                )
-
-                guard result == .completed else { throw WaitForScreenError.timedOut }
+        var currentRetryCount = 0
+        while currentRetryCount < maxRetries {
+            currentRetryCount += 1
+            do {
+                try XCTContext.runActivity(named: activityDescription) { (activity) in
+                    try gettersToTest.forEach { getter in
+                        let result = waitFor(
+                            element: getter(app),
+                            predicate: "isHittable == true",
+                            timeout: self.waitTimeout
+                        )
+
+                        guard result == .completed else { throw WaitForScreenError.timedOut }
+                    }
+                }
+                break
+            } catch {
+                if currentRetryCount < maxRetries {
+                    // Wait 1 second before retrying
+                    sleep(1)
```

Comment:
Is the sleep necessary when we already wait for `waitTimeout`?

Reply:
If the sleep is necessary, I'd suggest making it configurable.

Reply:
in this case, i think it's necessary for that short wait before retrying again and should be a really low number. i'll make it configurable